lib/dialog-renderer/index.tsx

```diff
@@ -1,64 +1,96 @@
-import React, { Fragment } from 'react';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
 import Modal from 'react-modal';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import { get } from 'lodash';
 
+import DialogTypes from '../../shared/dialog-types';
 import * as Dialogs from '../dialogs';
+import { closeDialog } from '../state/ui/actions';
+
+import * as S from '../state';
+import * as T from '../types';
+
+type StateProps = {
+  dialogs: T.DialogType[];
+};
+
+type DispatchProps = {
+  closeDialog: () => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+export class DialogRenderer extends Component<Props> {
+  static displayName = 'DialogRenderer';
+
+  static propTypes = {
+    appProps: PropTypes.object.isRequired,
+    buckets: PropTypes.object,
+    themeClass: PropTypes.string,
+    isElectron: PropTypes.bool.isRequired,
+    isMacApp: PropTypes.bool.isRequired,
+  };
+
+  renderDialog(dialogType: string) {
+    const {
+      appProps,
+      buckets,
+      themeClass,
+      isElectron,
+      isMacApp,
+      closeDialog,
+    } = this.props;
+
+    const dialog = get(DialogTypes, dialogType);
+
+    if (dialog === null) {
+      throw new Error('Unknown dialog type.');
+    }
 
-export const DialogRenderer = props => {
-  const {
-    appProps,
-    buckets,
-    themeClass,
-    closeDialog,
-    dialogs,
-    isElectron,
-    isMacApp,
-  } = props;
-
-  const renderDialog = dialog => {
-    const { key, title } = dialog;
     const DialogComponent = Dialogs[dialog.type];
 
     if (DialogComponent === null) {
       throw new Error('Unknown dialog type.');
     }
 
-    const closeThisDialog = () => closeDialog({ key });
-
     return (
       <Modal
-        key={key}
+        key={dialog.title}
         className="dialog-renderer__content"
-        contentLabel={title}
+        contentLabel={dialog.title}
         isOpen
-        onRequestClose={closeThisDialog}
+        onRequestClose={closeDialog}
         overlayClassName="dialog-renderer__overlay"
         portalClassName={classNames('dialog-renderer__portal', themeClass)}
       >
         <DialogComponent
           buckets={buckets}
           dialog={dialog}
-          requestClose={closeThisDialog}
           isElectron={isElectron}
           isMacApp={isMacApp}
           {...appProps}
         />
       </Modal>
     );
-  };
+  }
 
-  return <Fragment>{dialogs.map(renderDialog)}</Fragment>;
-};
+  render() {
+    return (
+      <Fragment>{this.props.dialogs.map(this.renderDialog, this)}</Fragment>
+    );
```

Comment:
It seems like our rendering function isn't doing much and in general we can inline it here instead of abstracting it into a function.

```ts
render() {
  const { appProps, buckets, dialogs, isElectron, isMacApp, themeClass } = this.props;

  return (
    <Fragment>
      { dialogs.map( dialog => (
        <Modal … >
          { 'About' === dialog.type ? (
            <AboutDialog /> // no props needed
          ) : 'Import' === dialog.type ? (
            <ImportDialog buckets={buckets} isElectron={isElectron} />
          ) : 'Settings' === dialog.type ? (
            <SettingsDialog buckets={buckets} isElectron={isElectron} isMacApp={isMacApp} {...appProps} />
          ) : 'Share' === dialog.type ? (
            <ShareDialog /> // also I don't think any props are needed here
          ) : null
          }
        </Modal>
      ) ) }
    </Fragment>
  );
}
```

This also seems to suggest that the `dialogs` objects are unnecessary and your new `DialogType` should suffice as a string itself. The only use that I saw was that the dialogs render their title as `dialog.title`, which doesn't make sense since we have no other title for the `About` dialog than what it itself wants to render. Might as well take that out.